nilearn/regions/tests/test_parcellations.py

```diff
@@ -310,3 +311,26 @@ def test_transform_3d_input_images(affine_eye):
     X = parcellate.fit_transform(imgs[0])
     assert isinstance(X, np.ndarray)
     assert X.shape == (1, 20)
+
+
+@pytest.mark.parametrize("method", METHODS)
+@pytest.mark.parametrize("n_parcels", [2, 4, 5])
+def test_parcellation_all_methods_with_surface(
+    surf_img, surf_mask, method, n_parcels
+):
+    """Test if all parcellation methods work on surface."""
+    # create a surface masker
+    masker = SurfaceMasker(surf_mask()).fit()
+    # mask the surface image with 50 samples
+    X = masker.transform(surf_img((50,)))
+    parcellate = Parcellations(method=method, n_parcels=n_parcels)
+    # fit and transform the data
+    X_transformed = parcellate.fit_transform(X)
+    # inverse transform the transformed data
+    X_inverse = parcellate.inverse_transform(X_transformed)
+
+    # make sure the n_features in transformed data were reduced to n_clusters
+    assert X_transformed.shape[1] == n_parcels
+
+    # make sure the inverse transformed data has the same shape as the original
+    assert X_inverse.shape == X.shape
```

Comment:
@bthirion @Remi-Gau 

I think we forgot to fix [`Parcellations()`](https://github.com/nilearn/nilearn/blob/135e5388386c24ca879328040273127f8acb4a6a/nilearn/regions/parcellations.py#L120) class to work with surface data.

This test will fail because of that.

Reply:
ouch
do you want to look into it?

Reply:
Yes doing that right now

Reply:
This Parcellations class does masking before running the dimensionality reduction, and that is where we have the issue.

I think we need to update the [`_BaseDecomposition`]( https://github.com/nilearn/nilearn/blob/135e5388386c24ca879328040273127f8acb4a6a/nilearn/decomposition/_base.py#L247) object for this. This would involve implementing a new `SurfaceMapsMasker` object.

As discussed with @Remi-Gau, we could just merge the current PR because that would allow all Decoders to work with surface data. The current PR fixes the FREM classifier and regressor specifically.

So I could drop a NotImplemented error in case we or some users think that Parcellations class is supported, and then create a separate PR for Parcellations

Reply:
Whether or not we want to wait for the new `SurfaceMapsMasker` object before doing the next release is an open question for discussion...

Reply:
Ok, so I will just start the PR to create `SurfaceMapsMasker` and a separate PR for allowing Parcellations to work with surface API.

Reply:
I would prefer to keep the scope as it was for the coming release:
- get rid of experimental
- do not add new maskers
- therefore let's not support Parcellation on SurfaceImage

I would prefer having another release soonish after that to implement those new features and fix all the bugs that will come with users starting to use 0.11.

Reply:
I don't feel strongly about whether or not to delay the release to wait for `SurfaceMapsMasker`.
In any case, if we are ok with the reviews, let's merge this PR.

As @Remi-Gau suggested previously, I added `NotImplementedError` when trying to use surface data with Parcellations.